packages/jmix-react-ui/src/ui/table/DataTableIntervalFunctions.ts

```diff
@@ -1,7 +1,33 @@
-import moment, {Moment} from 'moment';
+import dayjs, {Dayjs} from 'dayjs';
 import {TemporalInterval, DataTableIntervalEditorMode, PredefinedIntervalOption, TimeUnit} from './DataTableIntervalEditor';
 import {PropertyType} from '@haulmont/jmix-rest';
-import {getDataTransferFormat} from '@haulmont/jmix-react-core';
+import {applyDataTransferFormat} from '@haulmont/jmix-react-core';
+
+const selectMinDateForInterval = (
+  mode: DataTableIntervalEditorMode,
+  numberOfUnits: number,
+  timeUnit: TimeUnit,
+  includeCurrent: boolean,
+): Dayjs => {
+  if(mode === 'last' && includeCurrent) return dayjs().subtract(numberOfUnits - 1, timeUnit).startOf(timeUnit)
+  else if(mode === 'last' && !includeCurrent) return dayjs().subtract(numberOfUnits, timeUnit).startOf(timeUnit)
+  else if(mode === 'next' && includeCurrent) return dayjs().add(0, timeUnit).startOf(timeUnit)
+  else if(mode === 'next' && !includeCurrent) return dayjs().add(1, timeUnit).startOf(timeUnit)
+  else throw new Error(`Expected 'last' or 'next' mode, encountered '${mode}' mode`)
```

Comment:
We need to include to CONTRIBUTING.md that our code style requires curly braces and `;`. As a personal preference, I'd also not to use `else`:

```
if (...) {
  return ...;
}

if (...) {
  return ...;
}
```